tedana/decomposition/gift_pca.py

```diff
@@ -0,0 +1,599 @@
+            parzen_w_2[(n[1] - M[1] - 1):(n[1] +
+                                          M[1])] = _parzen_win(2 * M[1] + 1)
+
+        if (x.ndim >= 1):
+            parzen_w_1 = np.zeros((2 * n[0] - 1, ))
+            parzen_w_1[(n[0] - M[0] - 1):(n[0] +
+                                          M[0])] = _parzen_win(2 * M[0] + 1)
+
+    if x.ndim == 2 and min(n) == 1:  # 1D
+        xc = _autocorr(x)
+        xc = xc * parzen_w_1
+        xf = fftshift(fft(xc))
+
+    elif x.ndim == 2 and min(n) != 1:  # 2D
+        xc = _autocorr(x)  # default option: computes raw correlations with NO
+        # normalization -- Matlab help on xcorr
+
+        # Bias correction
+        v1 = np.hstack((np.arange(1, n[0] + 1), np.arange(n[0] - 1, 0,
+                                                          -1)))[np.newaxis, :]
+        v2 = np.hstack((np.arange(1, n[1] + 1), np.arange(n[1] - 1, 0,
+                                                          -1)))[np.newaxis, :]
+
+        vd = np.dot(v1.T, v2)
+        xc = xc / vd
+        parzen_window_2D = np.dot(parzen_w_1, parzen_w_2.T)
+        xc = xc * parzen_window_2D
+        xf = fftshift(fft2(xc))
+
+    elif x.ndim == 3 and min(n) != 1:  # 3D
+        xc = np.zeros((2 * n[0] - 1, 2 * n[1] - 1, 2 * n[2] - 1))
+        for m3 in range(n[2] - 1):
+            temp = np.zeros((2 * n[0] - 1, 2 * n[1] - 1))
+            for k in range(n[2] - m3):
+                temp = temp + correlate2d(x[:, :, k + m3], x[:, :, k])
+                # default option:
+                # computes raw correlations with NO normalization
+                # -- Matlab help on xcorr
+            xc[:, :, (n[2] - 1) - m3] = temp
+            xc[:, :, (n[2] - 1) + m3] = temp
+
+        # Bias correction
+        v1 = np.hstack((np.arange(1, n[0] + 1), np.arange(n[0] - 1, 0,
+                                                          -1)))[np.newaxis, :]
+        v2 = np.hstack((np.arange(1, n[1] + 1), np.arange(n[1] - 1, 0,
+                                                          -1)))[np.newaxis, :]
+        v3 = np.arange(n[2], 0, -1)
+
+        vd = np.dot(v1.T, v2)
+        vcu = np.zeros((2 * n[0] - 1, 2 * n[1] - 1, 2 * n[2] - 1))
+        for m3 in range(n[2]):
+            vcu[:, :, (n[2] - 1) - m3] = vd * v3[m3]
+            vcu[:, :, (n[2] - 1) + m3] = vd * v3[m3]
+
+        # Possible source of NAN values
+        xc = xc / vcu
+
+        parzen_window_2D = np.dot(parzen_w_1[np.newaxis, :].T,
+                                  parzen_w_2[np.newaxis, :])
+        parzen_window_3D = np.zeros((2 * n[0] - 1, 2 * n[1] - 1, 2 * n[2] - 1))
+        for m3 in range(n[2] - 1):
+            parzen_window_3D[:, :, (n[2] - 1) - m3] = np.dot(
+                parzen_window_2D, parzen_w_3[n[2] - 1 - m3])
+            parzen_window_3D[:, :, (n[2] - 1) + m3] = np.dot(
+                parzen_window_2D, parzen_w_3[n[2] - 1 + m3])
+
+        xc = xc * parzen_window_3D
+
+        xf = fftshift(fftn(xc))
+
+    else:
```

Comment:
Yeah I _really_ don't see how we would hit this..... Can you describe a case ?

Reply:
It can only happen if `vcu = 0`

Reply:
I'd keep it just in case